### PR TITLE
[WIP] $brooklyn:self().template("freemarker")

### PR DESCRIPTION
**Pull request by neykov**

### CODE REVIEW

File: camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/parse/DslParser.java [hunk 1]
```diff
@@ -52,7 +53,29 @@ public Object next() {
         if (index >= expression.length())
             throw new IllegalStateException("Unexpected end of expression to parse, looking for content since position "+start);
         
-        if (expression.charAt(index)=='"') {
+        String multilineTemplatePrefix = "$brooklyn:template";
```
Comment (geomacy) [inline camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/parse/DslParser.java]: Seems odd to have just `$brooklyn:template` and then the line break - can this not be done something like
```
$brooklyn:template: |
   my ${template} here
```

File: camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/parse/DslParser.java [hunk 2]
```diff
@@ -52,7 +53,29 @@ public Object next() {
...
+            // skip to next line
+            char c;
+            do {
+                c = expression.charAt(index);
+                index++;
+            } while(c != '\n' && index<expression.length());
+            if (index < expression.length()) {
+                int stringStart = index;
+                index = expression.length();
+                return new QuotedString('"' + JavaStringEscapes.escapeJavaString(expression.substring(stringStart, index)) + '"');
```
Comment (geomacy) [inline camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/parse/DslParser.java]: Need to skip trailing newline/whitespace on the expression: I tried updating the test to add further text:

```java
    protected Entity entityWithMultilineTemplatedString() throws Exception {
        return setupAndCheckTestEntityInBasicYamlWith(
                "  id: x",
                "  brooklyn.config:",
                "    test.sourceName: hello world",
                "    test.confName: |",
                "      $brooklyn:template",
                "      ${config['test.sourceName']}",
                "    test.other: something else");
    }
```

and it breaks with 
```
java.lang.AssertionError: expected [hello world] but found [hello world
]
```

File: core/src/test/java/org/apache/brooklyn/util/core/text/TemplateProcessorTest.java [hunk 3]
```diff
@@ -71,6 +71,15 @@ public void testEntityConfig() {
     }
     
     @Test
+    public void testDsl() {
+        TestEntity entity = app.createAndManageChild(EntitySpec.create(TestEntity.class)
+                .configure(TestEntity.CONF_NAME, "myval"));
+        String templateContents = "${brooklyn(\"id\")}";
```
Comment (geomacy) [inline core/src/test/java/org/apache/brooklyn/util/core/text/TemplateProcessorTest.java]: Surely should be 
```java
        String templateContents = "${config['" + TestEntity.CONF_NAME.getName() + "']}";
```
As written this test breaks for me.

**Comment by neykov** [comment]
Closing in favour of https://github.com/apache/brooklyn-server/pull/786.

**Pull request closed without merging**